stratum/hal/lib/common/phaldb_service_test.cc

```diff
@@ -0,0 +1,861 @@
+    server_ = builder.BuildAndStart();
+    ASSERT_NE(server_, nullptr);
+    stub_ = ::stratum::hal::phal::PhalDBSvc::NewStub(
+        ::grpc::CreateChannel(url, ::grpc::InsecureChannelCredentials()));
+    ASSERT_NE(stub_, nullptr);
+
+    database_mock_ =
+        absl::make_unique<::stratum::hal::phal::AttributeDatabaseMock>();
+  }
+
+  void TearDown() override {
+    phaldb_service_->Teardown();
+    server_->Shutdown();
+  }
+
+  OperationMode mode_;
+  std::unique_ptr<AuthPolicyCheckerMock> auth_policy_checker_mock_;
+  std::unique_ptr<ErrorBuffer> error_buffer_;
+  std::unique_ptr<PhalDBService> phaldb_service_;
+  std::unique_ptr<::grpc::Server> server_;
+  std::unique_ptr<::stratum::hal::phal::PhalDBSvc::Stub> stub_;
+  std::unique_ptr<PhalMock> phal_mock_;
+  std::unique_ptr<::stratum::hal::phal::AttributeDatabaseMock> database_mock_;
+};
+
+TEST_P(PhalDBServiceTest, SetupWarm) {
+  auto status = phaldb_service_->Setup(true);
+  EXPECT_TRUE(status.ok());
+}
+
+TEST_P(PhalDBServiceTest, GetRequestStrSuccess) {
+  ::grpc::ClientContext context;
+  ::stratum::hal::phal::GetRequest req;
+  ::stratum::hal::phal::GetResponse resp;
+
+  // Returned PhalDB
+  auto phaldb_resp = absl::make_unique<::stratum::hal::phal::PhalDB>();
+  auto card = phaldb_resp->add_cards();
+  auto port = card->add_ports();
+  port->set_physical_port_type(PHYSICAL_PORT_TYPE_SFP_CAGE);
+  auto sfp = port->mutable_transceiver();
+  sfp->set_id(0);
+  sfp->set_description("port-0");
+  sfp->set_hardware_state(HW_STATE_PRESENT);
+  sfp->set_media_type(MEDIA_TYPE_SFP);
+  sfp->set_connector_type(SFP_TYPE_SFP);
+  sfp->set_module_type(SFP_MODULE_TYPE_10G_BASE_CR);
+  sfp->mutable_info()->set_mfg_name("test vendor");
+  sfp->mutable_info()->set_part_no("test part #");
+  sfp->mutable_info()->set_serial_no("test1234");
+
+  // Create mock query
+  auto db_query_mock = absl::make_unique<::stratum::hal::phal::QueryMock>();
+  // Need to get pointer before it gets moved
+  auto db_query = db_query_mock.get();
+
+  // Add Path
+  std::vector<::stratum::hal::phal::Path> paths = {{
+    ::stratum::hal::phal::PathEntry("cards", 0),
+    ::stratum::hal::phal::PathEntry("ports", 0),
+    ::stratum::hal::phal::PathEntry("transceiver", -1, false, false, true)
+  }};
+
+  // Setup Mock DB calls
+  auto phal_ptr = phal_mock_.get();
+  auto database = database_mock_.get();
+  EXPECT_CALL(*phal_ptr, GetPhalDB()).WillRepeatedly(Return(database));
+
+  EXPECT_CALL(*database, DoMakeQuery(_))
+      .WillOnce(Return(ByMove(std::move(db_query_mock))));
+
+  EXPECT_CALL(*db_query, DoGet())
+    .WillOnce(Return(ByMove(std::move(phaldb_resp))));
+
+  req.set_str("cards[0]/ports[0]/transceiver/");
+
+  EXPECT_CALL(*auth_policy_checker_mock_, Authorize("PhalDBService", "Get", _))
+      .WillOnce(Return(::util::OkStatus()));
+
+  // Invoke the RPC and validate the results.
+  // Call and validate results.
+  auto status = stub_->Get(&context, req, &resp);
+  ASSERT_TRUE(status.ok());
```

Comment:
We can use `EXPECT_OK` for these kinds of checks, same as below

Reply:
done